camel-core/src/main/java/org/apache/camel/processor/Throttler.java

```diff
@@ -192,6 +217,26 @@ public boolean process(final Exchange exchange, final AsyncCallback callback) {
         }
     }
 
+    private DelayQueue<ThrottlePermit> locateDelayQueue(final Integer key) throws InterruptedException, ExecutionException {        
```

Comment:
Could you please elaborate on why you are using a CompletableFuture here ?

Reply:
I could call complete manually, to get the result. That was it.